scripts/prepare-subcharts

```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/env bash
+set -e
+
+# Pull in the subcharts required by a chart and apply any patches on subcharts with fixed version numbers
+
+if [[ -z $1 ]]; then
+	echo "No directory provided to pull in subcharts"
+	exit 1
+fi
+
+f=$1
+
+if ! [[ -f ${f}/charts/requirements.yaml ]]; then
+	echo "Chart ${CHART} has no dependencies; skipping step to pull in dependencies"
+	exit 0
+fi
+
+# Apply the patch on the requirements.yaml
+for file in $(find ./${f} -type f -name "*.patch"); do
+	awk '/diff.*requirements.yaml/{flag=1;next}/diff/{flag=0}flag' $file >> ${f}/requirements.patch
+done
+if [[ -f ${f}/requirements.patch ]] && [[ -s ${f}/requirements.patch ]]; then
+	patch -E -p3 -d ${f}/charts < ${f}/requirements.patch
+fi
+
+# Pull in any dependencies that are stored as package.yamls or in the overlay/ folder
+overlaid_dependencies=()
+mirrored_dependencies=()
+i=0
+while ! [[ -z $(yq r ${f}/charts/requirements.yaml "dependencies[${i}]") ]]; do
+	repo=$(yq r ${f}/charts/requirements.yaml "dependencies[${i}].repository")
+	if [[ ${repo} == file://* ]]; then
+		charts_path=${repo/file:\/\//${f}/charts/}
+		if [[ -d ${charts_path} ]]; then
+			# Chart already exists
+			((i=i+1))
+			continue
+		fi
+		if [[ -f ${charts_path%/*}/package.yaml ]]; then
+			# Prepare the subchart and pull it in
+			subchart=$(basename -- ${charts_path%/*})
+			mirrored_dependencies+=("${subchart}")
+			prepare_subchart () {
+				local CURRENT_CHART=${CHART}
+				local CURRENT_INDEX=${i}
+				CHART=${subchart} ./scripts/prepare
+				CHART=${CURRENT_CHART}
+				i=${CURRENT_INDEX}
+			}
+			prepare_subchart
+		fi
+		overlay_path=${repo/file:\/\//${f}/overlay/}
+		if [[ -d $overlay_path ]]; then
+			# Pull in the subchart from the overlay/ directory
+			cp -R $overlay_path $charts_path
+			overlaid_dependencies+=("${charts_path}")
+		fi
+	fi
+	((i=i+1))
+done
+
+# Update the current requirements.lock and requirements.yaml to keep track of the overlay/ dependencies
+pwd=$(pwd)
+cd ${f}/charts
+helm dependency update
+cd $pwd
+
+# Remove any dependencies that came from the overlay/ directory
+for dep in "${overlaid_dependencies[@]}"; do
+	rm -rf $dep
+done
+
+# Cleanup any dependencies that came from mirrored package.yamls
```

Comment:
This is only for local charts that have package.yaml right ? 

```suggestion
# Cleanup any dependencies that came from another local chart, any changes should be made there. 
```

Reply:
Yes, this is intended to clean up charts like `rancher-kiali-server` that we ran a prepare for when trying to prepare `rancher-istio` only.

Will update the comment like you mentioned.